src/hotspot/share/prims/stackwalk.cpp

```diff
@@ -544,7 +544,11 @@ jint StackWalk::fetchNextBatch(Handle stackStream, jint mode, jlong magic,
     // the continuation and it returns to let Java side set the continuation.
     // Now this batch starts right at the first frame of another continuation.
     if (last_batch_count > 0) {
-      log_debug(stackwalk)("advanced past %s", stream.method()->external_name());
+      // It is not always safe to dig out the name of the last frame
+      // here, i.e. stream.method()->external_name(), since it may
+      // have been reclaimed by HandleMark::pop_and_restore() together
+      // with the rest of the previous batch.
+      log_debug(stackwalk)("advanced past last frame decoded in the previous batch");
```

Comment:
Your description says ResourceMark but this comment says HandleMark.  Stream is native memory allocated as a secret field of the frames_array somewhere (resource allocated?), and frames_array is a Handle subject to HandleMark.  I don't understand which got deallocated with this comment.

Reply:
Sorry I mixed up things in the description. But HandleMark is correct. There are two stack backtraces in [JDK-8325153](https://bugs.openjdk.org/browse/JDK-8325153) that show the actual details.